YAKL_init.h

```diff
@@ -3,14 +3,26 @@
 
 #include "YAKL_alloc_free.h"
 
-
   // Initialize the YAKL framework
   inline void init() {
     bool use_pool = true;
 
-    #ifdef __USE_SYCL__
+    #ifdef __USE_SYCL_
```

Comment:
This and following keeps kernels asynchronous, but forces them to be done in-order because we're using pointers, not Buffers.

This will trap async errors and report and die.